src/test/java/org/primeframework/mvc/cors/CORSFilterTest.java

```diff
@@ -217,6 +255,107 @@ public void options() throws Exception {
+                   .header("Access-Control-Request-Method", "GET")
+                   .header("Origin", "https://jackinthebox.com")
+                   .build(),
+        BodyHandlers.discarding()
+    );
+
+    // assert
+    var expectedStatus = expectCorsAllow ? 204 : 403;
+    assertEquals(response.statusCode(), expectedStatus);
+  }
+
+  @Test(dataProvider = "get_included_path_pattern")
+  public void options_included_path_pattern(String path, boolean expectCorsAllow) throws Exception {
+    // arrange
+    corsConfiguration = new CORSConfiguration().withAllowCredentials(true)
+                                               .withAllowedMethods(HTTPMethod.GET, HTTPMethod.POST, HTTPMethod.HEAD, HTTPMethod.OPTIONS, HTTPMethod.PUT, HTTPMethod.DELETE)
+                                               .withAllowedHeaders("Accept", "Access-Control-Request-Headers", "Access-Control-Request-Method", "Authorization", "Content-Type", "Last-Modified", "Origin", "X-FusionAuth-TenantId", "X-Requested-With")
+                                               .withAllowedOrigins(URI.create("*"))
+                                               .withIncludedPathPattern(Pattern.compile("^/admin/foo"))
+                                               .withExposedHeaders("Access-Control-Allow-Origin", "Access-Control-Allow-Credentials")
+                                               .withPreflightMaxAgeInSeconds(1800);
+
+    // act
+    HttpClient client = HttpClient.newBuilder().followRedirects(Redirect.NEVER).priority(256).build();
+    HttpResponse<Void> response = client.send(
+        HttpRequest.newBuilder(URI.create("http://localhost:9080" + path))
+                   .method("OPTIONS", BodyPublishers.noBody())
+                   .header("Access-Control-Request-Method", "GET")
+                   .header("Origin", "https://jackinthebox.com")
+                   .build(),
+        BodyHandlers.discarding()
+    );
+
+    // assert
+    var expectedStatus = expectCorsAllow ? 204 : 403;
+    assertEquals(response.statusCode(), expectedStatus);
+    if (!expectCorsAllow) {
+      return;
+    }
+    assertEquals(response.headers().firstValue("Access-Control-Allow-Credentials").orElse(null), "true");
+    assertEquals(response.headers().firstValue("Access-Control-Allow-Headers").orElse(null), "Accept,Access-Control-Request-Headers,Access-Control-Request-Method,Authorization,Content-Type,Last-Modified,Origin,X-FusionAuth-TenantId,X-Requested-With");
+    assertEquals(response.headers().firstValue("Access-Control-Allow-Methods").orElse(null), "GET");
+    assertEquals(response.headers().firstValue("Access-Control-Max-Age").orElse(null), "1800");
+    assertEquals(response.headers().firstValue("Access-Control-Allow-Origin").orElse(null), "https://jackinthebox.com");
+    assertNull(response.headers().firstValue("Access-Control-Expose-Headers").orElse(null));
+    assertEquals(response.headers().firstValue("Vary").orElse(null), "Origin");
```

Comment:
Ideally, we'd use the test simulator for this which has header assertions.

Reply:
Same reason as other

Reply:
Resolved in latest code (converted entire existing test class to use simulator)